Tests/MongoSwiftTests/DNSSeedlistTests.swift

```diff
@@ -76,70 +76,66 @@ final class DNSSeedlistTests: MongoSwiftTestCase {
 
             let topologyWatcher = TopologyDescriptionWatcher()
 
-            // Enclose all of the potentially throwing code in `doTest`. Sometimes the expected errors come when
-            // parsing the URI, and other times they are not until we try to select a server.
-            func doTest() throws -> ConnectionString {
-                let opts: MongoClientOptions?
-                if requiresTLS {
-                    opts = MongoClientOptions(
-                        tlsAllowInvalidCertificates: true
-                    )
-                } else {
-                    opts = nil
-                }
-
-                return try self.withTestClient(testCase.uri, options: opts) { client in
+            let opts: MongoClientOptions?
+            if requiresTLS {
+                opts = MongoClientOptions(tlsAllowInvalidCertificates: true)
+            } else {
+                opts = nil
+            }
+            do {
+                try self.withTestClient(testCase.uri, options: opts) { client in
                     client.addSDAMEventHandler(topologyWatcher)
+
                     // try selecting a server to trigger SDAM
                     _ = try client.connectionPool.selectServer(forWrites: false)
-                    return try client.connectionPool.getConnectionString()
-                }
-            }
 
-            // "You MUST verify that an error has been thrown if error is present."
-            if testCase.error == true {
-                expect(try doTest()).to(throwError(), description: testCase.comment ?? "")
+                    // get resolved connection string after SDAM has been started.
+                    let connStr = try client.connectionPool.getConnectionString()
+
+                    guard testCase.error != true else {
+                        XCTFail("Expected error for test case \(testCase.comment ?? ""), got none")
+                        return
+                    }
+
+                    // "You MUST verify that the set of ServerDescriptions in the client's TopologyDescription
+                    // eventually matches the list of hosts."
+                    // This needs to be done before the client leaves scope to ensure the SDAM machinery
+                    // keeps running.
+                    expect(topologyWatcher.lastTopologyDescription?.servers.map { $0.address })
```

Comment:
now that the SDAM machinery is started in the background and is more async, a race was happening between this scope exiting and the full topology being discovered. To ensure the topology is fully discovered every time, I moved all the assertions into the same scope as the client.